src/inputs/Table/Table.js

```diff
@@ -168,21 +173,46 @@ export const Table = (props) => {
           buildErrorsCountLabel={buildErrorsPanelTitle}
         />
       )}
-      <div data-cy="grid" id="grid-container" style={dimensions} className={agTheme}>
+      <div data-cy="grid" className={agTheme}>
         {isReady && (
-          <AgGridReact
-            ref={gridRef}
-            undoRedoCellEditing={true}
-            rowDragManaged={true}
-            suppressDragLeaveHidesColumns={true}
-            allowDragFromColumnsToolPanel={true}
-            columnDefs={formattedColumns}
-            defaultColDef={defaultColDef}
-            columnTypes={columnTypes}
-            rowData={rows}
-            context={context}
-            stopEditingWhenCellsLoseFocus={true}
-          />
+          <>
+            <TableFilterRow isFullscreen={isFullscreen} setFullscreen={onClickFullscreenButton} />
+            <Box sx={dimensions}>
+              <AgGridReact
+                ref={gridRef}
+                undoRedoCellEditing={true}
+                rowDragManaged={true}
+                suppressDragLeaveHidesColumns={true}
+                allowDragFromColumnsToolPanel={true}
+                columnDefs={formattedColumns}
+                defaultColDef={defaultColDef}
+                columnTypes={columnTypes}
+                rowData={rows}
+                context={context}
+                stopEditingWhenCellsLoseFocus={true}
+              />
+            </Box>
+            <Dialog fullScreen open={isFullscreen} onClose={onClickFullscreenButton} className={agTheme}>
+              <Box p={4} sx={{ height: '100%' }}>
+                <TableFilterRow isFullscreen={isFullscreen} setFullscreen={onClickFullscreenButton} />
+                <Box sx={{ height: 'calc(100% - 32px)' }}>
+                  <AgGridReact
+                    ref={gridRef}
+                    undoRedoCellEditing={true}
+                    rowDragManaged={true}
+                    suppressDragLeaveHidesColumns={true}
+                    allowDragFromColumnsToolPanel={true}
+                    columnDefs={formattedColumns}
+                    defaultColDef={defaultColDef}
+                    columnTypes={columnTypes}
+                    rowData={rows}
+                    context={context}
+                    stopEditingWhenCellsLoseFocus={true}
+                  />
```

Comment:
Could you declare this `AgGridReact` element in a `const` (probably with a `useMemo`) to prevent duplicating 13 lines please ? :smile: